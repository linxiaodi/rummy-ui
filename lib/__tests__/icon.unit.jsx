import React from 'react';
import Icon from '../icons';
import { shallow, mount } from 'enzyme';

const clickFn = jest.fn()

describe('icon component', () => {
	// basic renderer
	it('test props name', () => {
		const comp = shallow(<Icon name='alibaba'/>);
		expect(comp).toMatchSnapshot();
	});
	// event
	it('trigger click event', () => {
		const comp = shallow(<Icon onClick={clickFn} name='wechat'/>);
		comp.simulate('click')
		expect(clickFn).toBeCalled()
	});
});
