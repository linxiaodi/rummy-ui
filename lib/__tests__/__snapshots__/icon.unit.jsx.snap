// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon component test props name 1`] = `
<i
  className="ru-icon ru-icon-alibaba"
/>
`;
